Objects/object.c

```diff
@@ -2134,6 +2134,13 @@ _PyTrash_end(PyThreadState *tstate)
 }
 
 
+int
+_PyTrash_cond(PyObject *op, void *dealloc)
```

Comment:
Hum, `void*` type might cause an issue on some platforms where function pointers don't have the same size void* pointers. Would you mind to use the `destructor` type instead?

Reply:
Ok. In fact, I use `PyTrash_cond(Pyobject *op, destructor *dealloc)` in my local firstly.
Why I decided use double void pointer in here? just it will make the marco looks simple(hide the conversion operation.)

Reply:
`destructor *` is the wrong type: it should be `destructor`.

Py_TRASHCAN_BEGIN_CONDITION() should cast its second argument with `(destructor)` since it's common that destructor functions use a different type for their first argument. Example:

```
static void
element_dealloc(ElementObject* self)
{
    /* bpo-31095: UnTrack is needed before calling any callbacks */
    PyObject_GC_UnTrack(self);
    Py_TRASHCAN_BEGIN(self, element_dealloc)
```

Reply:
oh, sorry. typo error.

Reply:
done.